source/MaterialXCore/Element.cpp

```diff
@@ -81,6 +87,108 @@ bool Element::operator!=(const Element& rhs) const
     return !(*this == rhs);
 }
 
+bool Element::isEquivalent(ConstElementPtr rhs, const ElementEquivalenceOptions& options,
+                           ElementEquivalenceResults* result) const
```

Comment:
I really like this latest change, where the passed object is simply a vector of `ElementEquivalenceResult` objects.

For the name of this vector, I might suggest `ElementEquivalenceResultVec`, which seems more clear than just pluralizing the name of the underlying class.

Although we don't often name vector types in MaterialX, I believe we've used the `Vec` suffix in cases where we do:

https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/source/MaterialXCore/Library.h#L55

Reply:
Yes, this is better / more consistent, I'll make the change.